scripts/sign-macos-pkg.sh

```diff
@@ -32,53 +64,59 @@ function clean_up {
     exit $STATUS
 }
 
-# Flags for codesign
-CODESIGN_OPTS=(
-    "--sign ${CODESIGN_ID}"
-    "--options runtime"
-    "--verbose=4"
-    "--force"
-)
-# Add extra flags provided via command line
-CODESIGN_OPTS+=(
-    ${CODESIGN_OPTS_EXTRA[@]}
-)
-
-# Setting MACOS_KEYCHAIN_FILE nd MACOS_KEYCHAIN_PASS is not required because
-# MACOS_CODESIGN_IDENT can be found in e.g. your login keychain.
-# Those would normally be specified only in CI.
-if [[ -n "${MACOS_KEYCHAIN_FILE}" ]]; then
-    if [[ -z "${MACOS_KEYCHAIN_PASS}" ]]; then
-        echo "Unable to unlock the keychain without MACOS_KEYCHAIN_PASS!" >&2
-        exit 1
-    fi
-    echo -e "\n### Storing original keychain search list..."
-    # We want to restore the normal keychains and ignore Jenkis created ones
-    ORIG_KEYCHAIN_LIST=$(security list-keychains | grep -v -e "^/private" -e "secretFiles" | xargs)
-    
-    # The keychain file needs to be locked afterwards
-    trap clean_up EXIT ERR
-
-    echo -e "\n### Adding keychain to search list..."
-    security list-keychains -s ${ORIG_KEYCHAIN_LIST} "${MACOS_KEYCHAIN_FILE}"
-    security list-keychains
-    echo -e "\n### Unlocking keychain..."
-    security unlock-keychain -p "${MACOS_KEYCHAIN_PASS}" "${MACOS_KEYCHAIN_FILE}"
+if [ "$METHOD" = "frameworks" ]; then
```

Comment:
Why is this a parameter? Why can't we just detect what is what?
And what is the actual difference?